src/main/java/at/hannibal2/skyhanni/features/misc/FrogMaskFeatures.kt

```diff
@@ -0,0 +1,141 @@
+import at.hannibal2.skyhanni.utils.RegexUtils.firstMatcher
+import at.hannibal2.skyhanni.utils.RenderUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderable
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SimpleTimeMark.Companion.asTimeMark
+import at.hannibal2.skyhanni.utils.SkyBlockTime
+import at.hannibal2.skyhanni.utils.TimeUtils.format
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object FrogMaskFeatures {
+    private val config get() = SkyHanniMod.feature.misc.frogMaskFeatures
+
+    private var display: Renderable? = null
+
+    private var lastWarning = SimpleTimeMark.farPast()
+
+    private val patternGroup = RepoPattern.group("misc.frogmask")
+
+    /**
+     * REGEX-TEST: §7Today's region: §aDark Thicket
+     */
+    private val activeRegionPattern by patternGroup.pattern(
+        "description.active",
+        "§7Today's region: (?<region>.+)",
+    )
+
+    /**
+     * REGEX-TEST:  §7⏣ §aSpruce Woods
+     */
+    private val currentAreaPattern by patternGroup.pattern(
+        "scoreboard.current",
+        " §7⏣ (?<area>.+)",
+    )
+
+    val FROG_MASK = "FROG_MASK".toInternalName()
+    private val frogMask by lazy { FROG_MASK.getItemStack() }
+
+    @HandleEvent(GuiRenderEvent.GuiOverlayRenderEvent::class, onlyOnIsland = IslandType.THE_PARK)
+    fun onRenderOverlay() {
+        if (!isEnabled()) return
+
+        config.displayPosition.renderRenderable(display, posLabel = "Frog Mask Display")
+    }
+
+    @HandleEvent(SecondPassedEvent::class, onlyOnIsland = IslandType.THE_PARK)
+    fun onSecondPassed() {
+        if (!isEnabled()) return
+        display = null
+
+        val helmet = InventoryUtils.getHelmet() ?: return
+        if (helmet.getInternalName() != FROG_MASK) return
+
+        activeRegionPattern.firstMatcher(helmet.getLore()) {
+            val currentRegion = group("region")
+
+            if (config.warning.enabled) lastWarning = handleWarning(currentRegion)
+            if (config.display) display = handleDisplay(currentRegion)
+        }
+    }
+
+    private fun handleWarning(currentRegion: String): SimpleTimeMark {
+        if (config.warning.warningType == WarningType.NEVER) return lastWarning
+
+        currentAreaPattern.firstMatcher(ScoreboardData.sidebarLinesFormatted) {
+            val needsToWarn =
+                group("area") != currentRegion && lastWarning.passedSince() > config.warning.cooldown.seconds
+
+            if (!needsToWarn) return lastWarning
+
+            when (config.warning.warningType) {
+                WarningType.BEING -> TitleManager.sendTitle("§cWrong Region!")
+                WarningType.FORAGING -> if (isForaging()) TitleManager.sendTitle("§cWrong Region!")
+                else -> return lastWarning
+            }
+
+            return SimpleTimeMark.now()
```

Comment:
instead of returning the last warning time, rather return void and set the new last warning time here